browser/tor/tor_profile_service_impl.cc

```diff
@@ -132,5 +200,10 @@ void TorProfileServiceImpl::NotifyTorLaunched(bool result, int64_t pid) {
     observer.OnTorLaunched(result, pid);
 }
 
+std::unique_ptr<net::ProxyConfigService>
+TorProfileServiceImpl::CreateProxyConfigService() {
+  proxy_config_service_ = new net::ProxyConfigServiceTor(tor::GetTorProxyURI());
```

Comment:
`std::make_unique`

Reply:
that would make the code a lot messier here because I want to keep a non-owned reference to the original pointed and then return the unique_ptr

Reply:
I would prefer the code below, but it's a minor
```
auto service = std::make_unique<net::ProxyConfigServiceTor>(tor::GetTorProxyURI());
proxy_config_service_ = service.get();
return std::move(service);
```